### PR TITLE
[RateLimiter][Security] More precisely document advanced rate limiter configuration

### PR DESCRIPTION
**Pull request by wouterj**
These additions are based on questions I received from @Seldaek while he was trying out the login throttling feature on a more advanced high availability set-up.

Previously, we documented the happy-path, but were not so good at documenting the more advanced settings. This PR adds more detail about locks, storage and implementing your custom login rate limiter.

### CODE REVIEW

File: rate_limiter.rst [hunk 1]
```diff
@@ -300,27 +354,146 @@ the :class:`Symfony\\Component\\RateLimiter\\Reservation` object returned by the
...
-                cache_pool: 'app.redis_cache'
-                # or define a service implementing StorageInterface to use a different
-                # mechanism to store the limiter information
-                storage_service: 'App\RateLimiter\CustomRedisStorage'
-                # the value is the name of any lock defined in your application
-                lock_factory: 'app.rate_limiter_lock'
+Storing Rate Limiter State: Caching
+-----------------------------------
+
+All rate limiter policies require to store the state of the rate limiter
```
Comment (Nyholm) [inline rate_limiter.rst:360]: Hm.. Either just say "their state" and let the parentheses do the explanation

```suggestion
All rate limiter policies require to store their state
```

Or, we can say 

> ..to store the state of the RateLimiter

File: rate_limiter.rst [hunk 2]
```diff
@@ -300,27 +354,146 @@ the :class:`Symfony\\Component\\RateLimiter\\Reservation` object returned by the
...
+                ],
+            ],
+        ]);
+
+.. note::
+
+    Instead of using the Cache component, you can also implement a custom
+    storage. Create a PHP class that implements the
+    :class:`Symfony\\Component\\RateLimiter\\Storage\\StorageInterface` and
+    set the ``storage_service`` setting of each limiter to the service ID
```
Comment (Nyholm) [inline rate_limiter.rst:429]:
```suggestion
    use the ``storage_service`` setting of each limiter to the service ID
```

Maybe?

**Review by Nyholm — APPROVED** [review]
Thank you!

**Review by maxhelias — APPROVED** [review]

**Comment by javiereguiluz** [comment]
Wouter, thanks for these nice additions! While merging I did some small rewords to turn passive voice into active voice (_"locks are used by rate limiters ..."_ --> _"rate limiters use locks ..."_) but the contents were perfect.

**Pull request merged**